learning_journal/views.py

```diff
@@ -2,35 +2,47 @@
 from pyramid.view import view_config
 
 from sqlalchemy.exc import DBAPIError
+from pyramid.httpexceptions import HTTPNotFound
+from pyramid.httpexceptions import HTTPFound
+from .forms import EntryCreateForm, EntryUpdateForm
+
 
 from .models import (
     DBSession,
     MyModel,
+    Entry,
     )
 
-
-@view_config(route_name='home', renderer='templates/mytemplate.pt')
-def my_view(request):
-    try:
-        one = DBSession.query(MyModel).filter(MyModel.name == 'one').first()
-    except DBAPIError:
-        return Response(conn_err_msg, content_type='text/plain', status_int=500)
-    return {'one': one, 'project': 'learning_journal'}
-
-
-conn_err_msg = """\
-Pyramid is having a problem using your SQL database.  The problem
-might be caused by one of the following things:
-
-1.  You may need to run the "initialize_learning_journal_db" script
-    to initialize your database tables.  Check your virtual
-    environment's "bin" directory for this script and try to run it.
-
-2.  Your database server may not be running.  Check that the
-    database server referred to by the "sqlalchemy.url" setting in
-    your "development.ini" file is running.
-
-After you fix the problem, please restart the Pyramid application to
-try it again.
-"""
+@view_config(route_name='home', renderer='templates/list.jinja2')
+def index_page(request):
+    entries = Entry.all()
+    return {'entries': entries}
+
+@view_config(route_name='detail', renderer='templates/detail.jinja2')
+def view(request):
+    this_id = request.matchdict.get('id', -1)
+    entry = Entry.by_id(this_id)
+    if not entry:
+        return HTTPNotFound()
+    return {'entry': entry}
+
+@view_config(route_name='action', match_param='action=create', renderer='templates/edit.jinja2')
```

Comment:
Nor is this one.  I think you must have a number of files on your local machine that have not been added to the repository.  Remember that when a new file is created, git needs to be told that you care about it.  You have to `git add <filename>` in order for it to be tracked by git at all.  At this point, if I check out your code and try to start it, it fails.  It will also fail when we try to push it to Heroku.  Let's get that fixed before tonight.